### ISSUE TITLE
make run fails: no suitable video mode found

### ISSUE THREAD

**Issue by pixlxip** [asterinas/asterinas#830]
I clone the repo, enter the container (I used `podman`; I'm not sure if that matters but I'm unable to install Docker), `make build` (looks successful), and then this is the output of `make run`:

```
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
make[3]: Nothing to be done for 'all'.
[Info] [normal] Forwarded QEMU guest port: 5593->22; 27669->8080
[Info] [test] Forwarded QEMU guest port: 44814->22; 24420->8080
[Info] [microvm] Forwarded QEMU guest port: 18161->22; 36996->8080
[Info] [iommu] Forwarded QEMU guest port: 9065->22; 14138->8080
    Updating crates.io index
    Updating git repository `https://github.com/asterinas/inherit-methods-macro`
    Updating git repository `https://github.com/asterinas/pod`
    Updating git repository `https://github.com/asterinas/trapframe-rs`
    Updating git repository `https://github.com/asterinas/xarray`
    Updating git repository `https://github.com/asterinas/inventory`
    Updating git repository `https://github.com/asterinas/rust-ctor`
   Compiling asterinas-osdk-bin v0.4.0 (/root/asterinas/target/osdk/base)
    Finished dev [unoptimized + debuginfo] target(s) in 1.93s
xorriso 1.5.4 : RockRidge filesystem manipulator, libburnia project.

Drive current: -outdev 'stdio:/root/asterinas/target/osdk/asterinas-osdk-bin.iso'
Media current: stdio file, overwriteable
Media status : is blank
Media summary: 0 sessions, 0 data blocks, 0 data,  234g free
BdsDxe: loading Boot0001 "UEFI QEMU DVD-ROM QM00005 " from PciRoot(0x0)/Pci(0x1F,0x2)/Sata(0x2,0xFFFF,0x0)
BdsDxe: starting Boot0001 "UEFI QEMU DVD-ROM QM00005 " from PciRoot(0x0)/Pci(0x1F,0x2)/Sata(0x2,0xFFFF,0x0)
WARNING: no console will be available to OS
error: no suitable video mode found.
```

It then exits.

Some info about my system:
Vanilla OS 2.0 Beta (That's why I can't install Docker)
Linux 6.7.9-amd64
I just cloned the Git a couple minutes ago

**Comment by pixlxip** [comment]
1. I am on AMD, yes
2. When following that, I get an error after running `cargo osdk run`. (`cargo osdk build seemed to be successful and I followed the instructions for installing):
```
 󰮄 apx-vso-pico  master ~/P/myos ~> cargo osdk run

    Updating git repository `https://github.com/asterinas/asterinas`
    Updating crates.io index
    Updating git repository `https://github.com/asterinas/inherit-methods-macro`
    Updating git repository `https://github.com/asterinas/pod`
    Updating git repository `https://github.com/asterinas/trapframe-rs`
    Updating git repository `https://github.com/asterinas/xarray`
   Compiling myos-osdk-bin v0.1.0 (/home/pixl_xip/Projects/myos/target/osdk/base)
    Finished dev [unoptimized + debuginfo] target(s) in 0.81s
xorriso 1.5.6 : RockRidge filesystem manipulator, libburnia project.

Drive current: -outdev 'stdio:/home/pixl_xip/Projects/myos/target/osdk/myos-osdk-bin.iso'
Media current: stdio file, overwriteable
Media status : is blank
Media summary: 0 sessions, 0 data blocks, 0 data,  213g free
Added to ISO image: directory '/'='/tmp/grub.23jPGZ'
xorriso : UPDATE :     342 files added in 1 seconds
Added to ISO image: directory '/'='/home/pixl_xip/Projects/myos/target/osdk/iso_root'
xorriso : UPDATE :     346 files added in 1 seconds
ISO image produced: 19832 sectors
Written to medium : 19832 sectors at LBA 0
Writing to 'stdio:/home/pixl_xip/Projects/myos/target/osdk/myos-osdk-bin.iso' completed successfully.

qemu-system-x86_64: -drive if=pflash,format=raw,unit=0,readonly=on,file=/usr/share/OVMF/OVMF_CODE.fd: Could not open '/usr/share/OVMF/OVMF_CODE.fd': No such file or directory
 󰮄 apx-vso-pico  2  master ~/P/myos ~> ls /usr/share/OVMF/OVMF_CO…
…VMF/OVMF_CODE_4M.fd           …VMF/OVMF_VARS_4M.fd
…VMF/OVMF_CODE_4M.ms.fd        …VMF/OVMF_VARS_4M.ms.fd
…VMF/OVMF_CODE_4M.secboot.fd   …VMF/OVMF_VARS_4M.snakeoil.fd
…VMF/OVMF_CODE_4M.snakeoil.fd
```
3. `multiboot` gave the same output but `linux` is slightly different: It actually brought me to GRUB. However, I can't navigate it. It also gave an error about not being able to load an image. I'll send a video here.
[Screencast from 2024-05-12 21-35-40.webm](https://github.com/asterinas/asterinas/assets/99521063/d48da8b2-84ba-413e-8cc3-6ae58e3cb86e)

**Comment by cqs21** [comment]
According to the AMD manual Volume.2: 

> If the guest is in PAE paging mode according to the registers just loaded and nested paging is not enabled, the processor will also read the four PDPEs pointed to by the newly loaded CR3 value; setting any reserved bits in the PDPEs also causes a #VMEXIT.

The Global Page Bit of PDPE is reserved, so we should not set this bit when setting the boot page table through multiboot.

**Comment by junyang-zh** [comment]
Fixed by #1011
